src/skin/qml/qmlwaveformoverview.cpp

```diff
@@ -0,0 +1,267 @@
+#include "skin/qml/qmlwaveformoverview.h"
+
+#include "mixer/basetrackplayer.h"
+#include "skin/qml/qmlplayerproxy.h"
+
+namespace {
+constexpr double kDesiredChannelHeight = 255;
+} // namespace
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+QmlWaveformOverview::QmlWaveformOverview(QQuickItem* parent)
+        : QQuickPaintedItem(parent),
+          m_pPlayer(nullptr),
+          m_channels(ChannelFlag::BothChannels),
+          m_renderer(Renderer::RGB),
+          m_colorHigh(0xFF0000),
+          m_colorMid(0x00FF00),
+          m_colorLow(0x0000FF) {
+}
+
+QmlPlayerProxy* QmlWaveformOverview::getPlayer() const {
+    return m_pPlayer;
+}
+
+void QmlWaveformOverview::setPlayer(QmlPlayerProxy* pPlayer) {
+    if (m_pPlayer == pPlayer) {
+        return;
+    }
+
+    if (m_pPlayer != nullptr) {
+        disconnect(m_pPlayer->internalTrackPlayer(), nullptr, this, nullptr);
```

Comment:
Why not just `m_pPlayer->internalTrackPlayer()->disconnect(this)`?

https://doc.qt.io/qt-5/qobject.html#disconnect-3

Reply:
good idea, didn't know this method.